packages/calcite-components/src/components/chip/chip.scss

```diff
@@ -25,28 +92,141 @@
+    --calcite-internal-chip-padding-end: var(--calcite-spacing-px);
+  }
+
+  .multiple {
+    .select-icon {
+      --calcite-internal-chip-padding-start: var(--calcite-spacing-base);
+      --calcite-internal-chip-padding-end: var(--calcite-spacing-base);
+    }
+
+    &.image--slotted {
+      .select-icon {
+        --calcite-internal-chip-padding-start: var(--calcite-spacing-xxs);
+        --calcite-internal-chip-padding-end: var(--calcite-spacing-sm);
+      }
+
+      &.text--slotted {
+        --calcite-internal-chip-padding-start: var(--calcite-spacing-xxs);
+        --calcite-internal-chip-padding-end: var(--calcite-spacing-xs);
+      }
+    }
+  }
+
+  .container {
+    --calcite-internal-chip-padding-start: var(--calcite-spacing-xs);
+    --calcite-internal-chip-padding-end: var(--calcite-spacing-xs);
+
+    &.image--slotted {
+      --calcite-internal-chip-padding-start: var(--calcite-spacing-xxs);
+      --calcite-internal-chip-padding-end: var(--calcite-spacing-xs);
+
+      &.closable {
+        --calcite-internal-chip-padding-start: var(--calcite-spacing-xxs);
+        --calcite-internal-chip-padding-end: var(--calcite-spacing-xxs);
+      }
+    }
+  }
+
+  .container:not(.is-circle):not(.multiple) {
+    &:not(.image-slotted) .select-icon.select-icon--active {
+      padding-inline-start: 0;
+      padding-inline-end: var(--calcite-spacing-px); // should be 2px, but xxxs doesn't work
+    }
+
+    &.image--slotted .select-icon.select-icon--active {
+      padding-inline-start: 0;
+      padding-inline-end: var(--calcite-spacing-sm);
+    }
   }
 }
 
+:host([scale="m"][closable]) .container {
+  --calcite-internal-chip-padding-start: var(--calcite-spacing-xs);
+  --calcite-internal-chip-padding-end: var(--calcite-spacing-xxs);
+}
+
 :host([scale="l"]) {
   @apply text-0 h-11;
+
   --calcite-chip-spacing-s-internal: theme("spacing[2]");
   --calcite-chip-spacing-l-internal: theme("spacing[3]");
+
   .image-container,
   .close,
   .select-icon--active {
     @apply h-8 w-8;
+
+    --calcite-internal-chip-padding-start: var(--calcite-spacing-xxs);
+    --calcite-internal-chip-padding-end: var(--calcite-spacing-xxs);
   }
 
-  .container.is-circle {
+  .container.is-circle,
+  .container.is-circle.image--slotted {
     @apply w-11 h-11;
+
+    --calcite-internal-chip-padding-start: var(--calcite-spacing-xxs);
+    --calcite-internal-chip-padding-end: var(--calcite-spacing-xxs);
+  }
+
+  .multiple {
+    .select-icon {
+      --calcite-internal-chip-padding-start: var(--calcite-spacing-xxs);
+      --calcite-internal-chip-padding-end: var(--calcite-spacing-xxs);
+    }
+
+    &.image--slotted {
+      .select-icon {
+        --calcite-internal-chip-padding-start: var(--calcite-spacing-xs);
+        --calcite-internal-chip-padding-end: var(--calcite-spacing-md);
+      }
+    }
+  }
+
+  .container {
+    --calcite-internal-chip-padding-start: var(--calcite-spacing-sm);
```

Comment:
Per our conversation, can you create a refactor issue to clean up chip's styling to reduce duplication and improve maintainability?

Reply:
[Refactor chip styling](https://github.com/Esri/calcite-design-system/issues/9785#top)
#9785